string-manipulation-3/src/title-case.js

```diff
@@ -0,0 +1,45 @@
+/* exported titleCase */
+
+function titleCase(string) {
+  var lower = string.toLowerCase();
+  var strArr = lower.split(' ');
+  for (var i = 0; i < strArr.length; i++) {
+    if (strArr[i] === 'javascript') {
+      strArr[i] = 'JavaScript';
+    } else if (strArr[i] === 'api') {
+      strArr[i] = 'API';
+    } else if (strArr[i] === 'web') {
+      var firstUp = strArr[i].charAt(0).toUpperCase();
+      var end = strArr[i].slice(1, strArr[i].length);
+      var finished = firstUp + end;
+      strArr.splice(i, 1, finished);
+    } else if (strArr[i].charAt(strArr[i].length - 1) === ':') {
+      var firstUpp = strArr[i + 1].charAt(0).toUpperCase();
+      var endd = strArr[i + 1].slice(1, strArr[i + 1].length);
+      var finishedd = firstUpp + endd;
+      strArr.splice(i + 1, 1, finishedd);
+      if (strArr[i] === 'javascript:') {
+        strArr[i] = 'JavaScript:';
+      }
+    } else if (strArr[i].includes('-')) {
+      var splitDash = strArr[i].split('-');
+      var firstUpper = splitDash[0].charAt(0).toUpperCase();
+      var secondUpper = splitDash[1].charAt(0).toUpperCase();
+      var firstEnd = splitDash[0].slice(1, splitDash[0].length);
+      var secondEnd = splitDash[1].slice(1, splitDash[1].length);
+      var firstFinished = firstUpper + firstEnd;
+      var secondFinished = secondUpper + secondEnd;
+      splitDash[0] = firstFinished;
+      splitDash[1] = secondFinished;
+      strArr[i] = splitDash.join('-');
+    } else if (strArr[i].length > 3) {
```

Comment:
This seems to work for this set of tests but would not always work, what if the title contained a name that is only 3 letters, (Joe, Sam, Tom, etc). It would be better to create an array of words that are exceptions to being capitalized, then check against that array to determine if the word should be capitalized.